docs/architecture/dapr-for-net-developers/publish-subscribe-buildingblock.md

```diff
@@ -0,0 +1,415 @@
+---
+title: The Dapr Publish/Subscribe building block
+description: A description of the Dapr Publish/Subscribe building-block and how to apply it
+author: edwinvw
+ms.date: 11/29/2020
+---
+
+# The Publish/Subscribe building block
+
+The [Publish-Subscribe pattern](https://docs.microsoft.com/azure/architecture/patterns/publisher-subscriber) is a well-known and widely used messaging pattern. Architects commonly embrace it in distributed applications. However, the plumbing to implement it can be complex. There are often subtle feature differences across different messaging products. Dapr offers a building block that significantly simplifies implementing publish/subscribe functionality.
+
+## What it solves
+
+The primary advantage of the Publish-Subscribe pattern is **loose coupling**, sometimes referred to as [temporal decoupling](https://docs.microsoft.com/azure/architecture/guide/technology-choices/messaging#decoupling). The pattern decouples services that send messages (the **publishers**) from services that consume messages (the **subscribers**). Both publishers and subscribers are unaware of each other - both are dependent on a centralized **message broker** that distributes the messages.
+
+Figure 7-1 shows the high-level architecture of the publish/subscribe pattern.
+
+![The Publish/Subscribe pattern](./media/pubsub-pattern.png)
+
+**Figure 7-1**. The publish/subscribe pattern.
+
+From the previous figure, note the steps of the pattern:
+
+- Publishers send messages to the message broker.
+- Subscribers bind to a subscription on the message broker.
+- The message broker forwards a copy of the message to interested subscriptions.
+- Subscribers consume messages from their subscriptions.
+
+Most message brokers encapsulate a queueing mechanism that can persist messages once received. With it, the message broker guarantees **durablity** by storing the message. Subscribers don't need to be immediately available or even online when a publisher sends a message.  Once available, the subscriber receives and processes the message.  Dapr guarantees **At-Least-Once** semantics for message delivery. Once a message is published, it will be delivered at least once to any interested subscriber.
```

Comment:
```suggestion
Most message brokers encapsulate a queueing mechanism that can persist messages once received. With it, the message broker guarantees **durability** by storing the message. Subscribers don't need to be immediately available or even online when a publisher sends a message.  Once available, the subscriber receives and processes the message.  Dapr guarantees **At-Least-Once** semantics for message delivery. Once a message is published, it will be delivered at least once to any interested subscriber.
```